mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -172,13 +176,15 @@ public RecordDaoImpl(final PostgresClientFactory postgresClientFactory) {
       .offset(offset)
       .limit(limit)
       .getSQL(ParamType.INLINED);
-    return getCachecPool(tenantId).rxBegin()
+
+    return getCachePool(tenantId)
```

Comment:
This is an incomplete conversion. This is no longer running in a transaction. Following the examples here https://vertx.io/docs/vertx-pg-client/java/#_streaming, it should be:

```
    return getCachedPool(tenantId)
      .rxGetConnection()
      .flatMapPublisher(conn -> conn.rxBegin()
        .flatMapPublisher(tx -> conn.rxPrepare(sql)
          .flatMapPublisher(pq -> pq.createStream(1)
            .toFlowable()
            .map(this::toRow)
            .map(this::toRecord))
          .doAfterTerminate(tx::commit)));
```

This affords to fetch the cursor with a single row and stream the remaining.

Reply:
@wwelling done